lm_eval/tasks/mlsum_de.py

```diff
@@ -0,0 +1,115 @@
+import datasets
+from lm_eval.base import rf, Task
+from functools import partial
+
+
+def _mlsum_metric(predictions, references):
+    summarization_metric = datasets.load_metric("rouge")
+    return summarization_metric.compute(predictions=predictions, references=references, use_agregator=True)
+
+
+def _mlsum_agg(key, items):
+    predictions, references = zip(*items)
+    return _mlsum_metric(predictions=predictions, references=references)[key].mid.fmeasure
+
+
+class MLSUM(Task):
+    VERSION = None
+    DATASET_PATH = "mlsum"
+    DATASET_NAME = 'de'
+
+    def has_training_docs(self):
+        return True
+
+    def has_validation_docs(self):
+        return True
+
+    def has_test_docs(self):
+        return True
+
+    def training_docs(self):
+        return self.data["train"]
+
+    def validation_docs(self):
+        return self.data["validation"]
+
+    def test_docs(self):
+        return self.dataset["test"]
+
+    # def fewshot_description(self):
+    #     return "Summarize the following articles."
+
+    def doc_to_text(self, doc):
+        return 'Article: ' + doc['text'] + '\n\n' + 'TL;DR:'
+
+    def doc_to_target(self, doc):
+        summary = doc['summary']
+        return " " + summary
+
+    def construct_requests(self, doc, ctx):
+        """ Uses RequestFactory to construct Requests and returns an iterable of
+        Requests which will be sent to the LM.
+
+        :param doc:
+            The document as returned from training_docs, validation_docs, or test_docs.
+        :param ctx: str
+            The context string, generated by fewshot_context. This includes the natural
+            language description, as well as the few shot examples, and the question
+            part of the document for `doc`.
+        """
+        continuation = rf.greedy_until(ctx, ['\n'], True, 2, 100) #arg3: do sampling, arg4: top k, arg5: max gen tokens
```

Comment:
Better use keyword args instead of putting the arg description into a line comment.

Reply:
Any change here requires change in the base.py, too. If you think it's fine, I can use keyword args.